src/imageio/storage/piwigo.c

```diff
@@ -706,8 +723,71 @@ static gboolean _piwigo_api_create_new_album(dt_storage_piwigo_params_t *p)
   return TRUE;
 }
 
+static char *_piwigo_api_get_image_id(dt_storage_piwigo_params_t *p, dt_image_t *img)
+{
+  GList *args = NULL;
+  char album_id[10];
+  sprintf(album_id, "%d", (int) p->album_id);
+
+  args = _piwigo_query_add_arguments(args, "method", "pwg.categories.getImages");
+  args = _piwigo_query_add_arguments(args, "cat_id", album_id);
+  args = _piwigo_query_add_arguments(args, "per_page", "999999");
```

Comment:
A bit simplistic :) Maybe worth having code to properly read multiple pages if needed until all data are returned. We are probably not sure the server will honor such a large page size!

Reply:
added paging by 100